gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/shared_policy_group/use_case/DeleteSharedPolicyGroupUseCase.java

```diff
@@ -49,7 +54,7 @@ public Output execute(Input input) {
     }
 
     @Builder
-    public record Input(String sharedPolicyGroupId, AuditInfo auditInfo) {}
+    public record Input(String sharedPolicyGroupId, String environmentId, AuditInfo auditInfo) {}
```

Comment:
You ca avoid adding new environmentId param by using the one already defined inside  auditInfo.  
Take the time to look at how things are done ;). Especially here, it's the new architecture that we want to stadardize.
Here, on line 57, we can see that we're already using the environmentId  from audit

Reply:
Well regarding "Take the time to look at how things are done ;)" - I was looking at `UndeploySharedPolicyGroupUseCase`, which defines both `environmentId` and `auditInfo` and seems to be more reasonable than reusing parts of `AuditInfo` for different purposes than audit.